lib/wallaroo/ent/router_registry/router_registry.pony

```diff
@@ -710,65 +714,75 @@ actor RouterRegistry is InFlightAckRequester
   //////////////
   // NEW WORKER PARTITION MIGRATION
   //////////////
-  be joining_worker_initialized(worker: String) =>
-    """
-    When a joining worker has initialized its topology, it contacts all
-    current workers. This behavior is called when that control
-    message is received.
-    """
-    if _joining_worker_count == 0 then
-      ifdef debug then
-        @printf[I32](("Joining worker reported as initialized, but we're " +
-          "not waiting for it. This should mean that either we weren't the " +
-          "worker contacted for the join or we're also joining.\n").cstring())
-      end
-      return
+  be report_connected_to_joining_worker(connected_worker: String) =>
+    try
+      (_autoscale as Autoscale).worker_connected_to_joining_workers(
+        connected_worker)
+    else
+      Fail()
     end
-    _initialized_joining_workers.set(worker)
-    if _initialized_joining_workers.size() == _joining_worker_count then
-      let nws = recover trn Array[String] end
-      for w in _initialized_joining_workers.values() do
-        nws.push(w)
-      end
-      let new_workers = consume val nws
-      _connections.notify_joining_workers_of_joining_addresses(new_workers)
-      try
-        let msg =
-          ChannelMsgEncoder.initiate_join_migration(new_workers, _auth)?
-        _connections.send_control_to_cluster(msg)
-      else
-        Fail()
-      end
-      // TODO: Where should this line go?
-      _initiated_stop_the_world = true
 
-      _migrate_onto_new_workers(new_workers)
-      _initialized_joining_workers.clear()
-      _joining_worker_count = 0
+  be remote_stop_the_world_for_join_migration_request(
+    joining_workers: Array[String] val)
+  =>
+    """
+    Only one worker is contacted by all joining workers to indicate that a
```

Comment:
The intro to this docstring appears here and in `remote_join_migration_request()`.  It is describing the coordinator, right, so perhaps add "coordinator" somewhere in both places?